[Profiling] Update the status of Allocated Memory; not available for .NET Framework (#29201)
DataDog/documentation: content/en/profiler/enabling/dotnet.md

| [Endpoint Profiling][13]  | 2.15.0+                            | All supported runtime versions.                                                          |
| Timeline                  | 2.30.0+ (and 3.10.0+ for outgoing HTTP requests longer than 50 ms in beta)     | All supported runtime versions (except .NET 5+ required for garbage collection details and .NET 7+ required for outgoing HTTP requests). |

> - Due to a limitation of .NET Framework, Allocations profiling do not show the size of the allocations but only their count.

Reviewer: ```suggestion
- Due to a limitation of the .NET Framework, Allocations profiling does not show the size of the allocations. Instead, it only shows the count.
```
